### PR TITLE
Ensure numeric data values are converted to the correct type

### PR DESCRIPTION
**Pull request by the-real-orca**
Fields that ate not known or used by ecowitt2mqtt are send as JSON string (e.g. winddir) while other values (like temp) are send as float. This creates problems further when processing the MQTT payload.
This PR fix the issue by converting all fields to their correct data types. Numerical values will be converted to JSON float, while text will still be send as string.

### CODE REVIEW

File: ecowitt2mqtt/main.py [hunk 1]
```diff
@@ -113,6 +115,10 @@ def main():
 
     LOGGER.debug("Using arguments: %s", args)
 
+    # workaround for windows python-3.8.0a4 NotImplementedError
```
Comment (bachya) [inline ecowitt2mqtt/main.py:118]: This is a wholly separate issue (Windows support) than what the PR states and should be in its own PR.

File: script/setup.bat [hunk 2]
```diff
@@ -0,0 +1,8 @@
+rem Install all dependencies:
```
Comment (bachya) [inline script/setup.bat:1]: This should be a separate PR.

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+wheel>=0.30.0
 aiohttp>=3.7.4
 asyncio-mqtt==0.7.0
 meteocalc==1.1.0
+pre-commit>=2.0.0
```
Comment (bachya) [inline requirements.txt:5]: Why?

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,5 @@
+wheel>=0.30.0
```
Comment (bachya) [inline requirements.txt:1]: Why?

**Review by bachya — CHANGES_REQUESTED** [review]
I'm unclear on the exact situation this PR is trying to solve – I've never had an issue with a numeric value not being presented properly. Can you share an example?

File: script/setup.bat (hunk 2 above)
Comment (the-real-orca) [reply script/setup.bat]: Yes, that's another commit and got into this PR. Have removed the Windows part.

**Comment by the-real-orca** [comment]
The problem is with values that are not used / processed by ecowitt2mqtt internally. Values like temp, etc. Are used internally and therefor converted correctly to floats. The other fields (e.g. humidity) are just passed through. This leads to a float encoded JSON value for temp and a string encoded JSON value for humidity.

original MQTT message: 
{"humidity": **"52"**, "temp": 12.3, ... }

patched  MQTT message:
{"humidity": **52.0**, "temp": 12.3, ... }

**Comment by bachya** [comment]
Thanks for the follow-up, @the-real-orca. I'm still unclear why this would be happening. To use your humidity example, the `DataProcessor`'s initialization logic should handle the conversion:


https://github.com/bachya/ecowitt2mqtt/blob/231c3d769ba834af5597c7b98370b39678baedb9/ecowitt2mqtt/data.py#L160-L163

We should figure out the issue upstream; that downstream logic should just be about final composition and presentation, not data manipulation.

**Comment by Advenoni** [comment]
I have the same problem as @the-real-orca here.
`{"humidityin": "56","soilmoisture1": "52","soilmoisture2": "66","soilbatt1": "1.6","soilbatt2": "1.6","baromabs": 961.5,"baromrel": 961.5,"tempin": 23.4}`
This is from a GW1000 with two WH51. The baromabs, baromrel and tempin are encoded as float, but the humidityin, soilmoisture and soilbatt are as string and not as float or int.

**Comment by bachya** [comment]
Thanks for the confirmation @Advenoni. After some investigation into my device, I failed to notice previously that I, too, am seeing this. Some numeric fields are converted correctly, some are not:

```json
{
  "humidityin": "36",
  "humidity": "26",
  "winddir": "25",
  "maxdailygust": 4.5,
  "solarradiation": "688.02",
  "uv": "6",
  "wh65batt": "0",
  "baromabs": 24.73,
  "baromrel": 24.73,
  "dailyrain": 0,
  "eventrain": 0,
  "hourlyrain": 0,
  "monthlyrain": 0.13,
  "rainrate": 0,
  "totalrain": 9.575,
  "weeklyrain": 1.571,
  "yearlyrain": 9.575,
  "temp": 88.9,
  "tempin": 82.2,
  "dewpoint": 49.6,
  "feelslike": null,
  "heatindex": 86,
  "windchill": null,
  "windgust": 1.1,
  "windspeed": 0
}
```

I still think @the-real-orca's proposal is too "catch-all" for my liking; we should determine where the issue is occurring upstream. I'll do some more research and propose next steps.

**Comment by bachya** [comment]
Okay, thanks for everyone's patience. I see now that there could be all sorts of different numeric fields coming through depending on the device; since we don't want to manually manage every possible field, @the-real-orca's solution works for me. I'll fix the linting issues and get this merged.
